relayer/connection-tx.go

```diff
@@ -95,87 +91,117 @@ func (c *Chain) CreateConnectionStep(dst *Chain) (*RelayMsgs, error) {
 		srcConsH, dstConsH               ibcexported.Height
 	)
 
-	// create the UpdateHeaders for src and dest Chains
-	eg.Go(func() error {
-		return retry.Do(func() error {
-			srcUpdateHeader, dstUpdateHeader, err = sh.GetTrustedHeaders(c, dst)
-			return err
-		}, rtyAtt, rtyDel, rtyErr, retry.OnRetry(func(n uint, err error) {
-			logRetryUpdateHeaders(c, dst, n, err)
-			sh.Updates(c, dst)
-		}))
-	})
-
-	// Query Connection data from src and dst
-	eg.Go(func() error {
-		srcConn, dstConn, err = QueryConnectionPair(c, dst, int64(sh.GetHeight(c.ChainID))-1, int64(sh.GetHeight(dst.ChainID))-1)
```

Comment:
pretty sure this was a bug. The retry callback above will update the sync headers if it fails, but this go routine, which presumably runs in parallel, needs the sync header height. It is safer to just do these calls synchronously instead of async. Would prefer to have a stable > fast implementation